### DISCUSSION TITLE
What would be the best way to use components? ( Buttons most importantly )

### DISCUSSION THREAD

**Discussion by StateGovernment** [bwmarrin/discordgo#1110] [General]
The current example file by itself doesn't make things clear to use Buttons or any other components, nor is there much documentation about components. Please provide some comments or some documentation regarding the same.

Amazing contribution so far, was a bliss to use DiscordGo.

**Comment by FedorLap2006** [comment]
Hi. Can you provde more details? — Which exact file? The example (`examples/components`) or `components.go` itself?

Additionally, documentation for components is [here](https://pkg.go.dev/github.com/bwmarrin/discordgo@master). It's not very obvious, because we don't have a tag yet, but it's going to change soon.

**Reply by StateGovernment** [reply]
I was referring to `examples/components`. Looking forward for the documentation. 😁😁😁

Thank you

**Reply by FedorLap2006** [reply]
> I was referring to `examples/components`. Looking forward for the documentation. gringringrin
> 
> Thank you

Which documentation? We're listed on pkg.go.dev

**Reply by StateGovernment** [reply]
I could see there is some documentation and an example regarding the same, but I couldn't make much sense on how to use them. So let me change the question to ask the community on how to use buttons in the best way.

**Reply by medzernik** [reply]
for example, I am trying to figure out a way to use the button as a confirmation mechanism, for instance to kill the bot remotely. Ideally, I want the button to be only pressable by the owner of the server, so it should be clickable only when the person interacting is the owner.

![image](https://user-images.githubusercontent.com/1900179/157088402-6abe7021-5ff6-4fb0-9be2-6654b1bd428d.png)

Also, I'd want to use the buttons as confirmations for several actions like this, or to have the bot suggest certain things (for example, if you want to mute someone, also suggesting a button to delete past messages from him.

**Reply by StateGovernment** [reply]
@medzernik You might be able to make the button be clickable to everyone but check whether or not the person clicked is an admin in the backend and kill the bot only if the user is an admin.

**Reply by medzernik** [reply]
@StateGovernment It already does the check like twice or thrice (just to be sure), since the bot is meant to be private hosted, but should work on more servers than one, meaning this kills it everywhere that the host set it up for.

I just thought its nicer to do the check like this, using the built-in discord features, the reply itself is only visible for the person who used the command anyway. 

Still thinking for more ideas though, especially for the new dropdowns and stuff. The buttons are really cool, though.

**Comment by odiferousmint** [comment]
While we are at it, how can I make the button disabled once clicked? When clicked, I send a message as per `components/main.go`, but I do not know how to make it disabled from there, if possible.

**Comment by Nv7-GitHub** [comment]
Use the `CustomID` field to differentiate between buttons. To disable a button once clicked, Edit the message with the `Components` field containing that button, but disabled.

**Reply by odiferousmint** [reply]
> Use the `CustomID` field to differentiate between buttons. To disable a button once clicked, Edit the message with the `Components` field containing that button, but disabled.

Oh thank you, that makes sense!

**Reply by Nv7-GitHub** [reply]
Can you mark as answer 🥺

**Reply by odiferousmint** [reply]
> Can you mark as answer pleading_face

I'm new to "Discussions". Is up-voting the way to do it?

**Reply by FedorLap2006** [reply]
@Nv7-GitHub @odiferousmint Just wanted to point out, this is a comment on another discussion.
You can only upvote it. Marking as an answer is only available in "Q&A".

**Reply by odiferousmint** [reply]
> @Nv7-GitHub @odiferousmint Just wanted to point out, this is a comment on another discussion. You can only upvote it. Marking as an answer is only available in "Q&A".

Yeah, that is what I thought. Thanks!

**Reply by Nv7-GitHub** [reply]
Ah ok that makes sense, trying to get the github badge
